### PR TITLE
Release @apollo/federation@0.31.0

### PR DESCRIPTION
**Pull request by trevor-scheer**
As with release PRs in the past, this is a PR tracking a release-federation-x.y.z branch for an upcoming release. 🙌 The version in the title of this PR should correspond to the appropriate branch.

The intention of these release branches is to gather changes which are intended to land in a specific version (again, indicated by the subject of this PR). Release branches allow additional clarity into what is being staged, provide a forum for comments from the community pertaining to the release's stability, and to facilitate the creation of pre-releases (e.g. alpha, beta, rc) without affecting the main branch.

PRs for new features might be opened against or re-targeted to this branch by the project maintainers. The main branch may be periodically merged into this branch up until the point in time that this branch is being prepared for release. Depending on the size of the release, this may be once it reaches RC (release candidate) stage with an -rc.x release suffix. Some less substantial releases may be short-lived and may never have pre-release versions.

When this version is officially released onto the latest npm tag, this PR will be merged into main.

### CODE REVIEW

File: gateway-js/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,10 @@
 
 - _Nothing yet! Stay tuned!_
 
+## v0.40.0
+
+- Only related changes in the `@apollo/federation` package. Adds support for @deprecated on input values and the new built-in directive @specifiedBy.
```
Comment (abernix) [inline gateway-js/CHANGELOG.md:11]: I'd suggest backtick the directives as code.

File: harmonizer/Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "harmonizer"
-version = "0.30.0"
+version = "0.31.0"
```
Comment (abernix) [inline harmonizer/Cargo.toml:3]: This requires a run of at least `cargo test` or `build` to update the lock file.

**Review by abernix — APPROVED** [review]
Couple comments lgtm. Thx!

File: harmonizer/Cargo.toml (hunk 2 above)
Comment (trevor-scheer) [reply harmonizer/Cargo.toml:3]: Good catch!

**Pull request merged**